src/components/transport_manager/src/tcp/tcp_socket_connection.cc

```diff
@@ -88,12 +92,21 @@ bool TcpServerOiginatedSocketConnection::Establish(ConnectError** error) {
     return false;
   }
 
+  if (is_connection_terminated()) {
+    LOG4CXX_ERROR(
+        logger_,
+        "Connection is already terminated. Socket will not be created");
+    *error = new ConnectError();
```

Comment:
@LuxoftAKutsan I think not, Connection Error  was just created in each "error" case of the Establish() method so I guess it contains default information.